interactive_engine/executor/ir/graph_proxy/src/adapters/gs_store/read_graph.rs

```diff
@@ -524,6 +549,154 @@ fn edge_trim(mut edge: Edge, columns: Option<&Vec<NameOrId>>) -> Edge {
     edge
 }
 
+pub struct HybridVertexDetails<V>
+where
+    V: StoreVertex + 'static,
+{
+    lazy: AtomicPtr<V>,
+    cached: Option<HashMap<PropId, Object>>,
+}
+
+impl<V> HybridVertexDetails<V>
+where
+    V: StoreVertex + 'static,
+{
+    pub fn new(v: V) -> Self {
+        let ptr = Box::into_raw(Box::new(v));
+        HybridVertexDetails { lazy: AtomicPtr::new(ptr), cached: None }
+    }
+
+    fn get_vertex_ptr(&self) -> Option<*mut V> {
+        let ptr = self.lazy.load(Ordering::SeqCst);
+        if ptr.is_null() {
+            None
+        } else {
+            Some(ptr)
+        }
+    }
+}
+
+impl<V> fmt::Debug for HybridVertexDetails<V>
+where
+    V: StoreVertex + 'static,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("gs_store LazyVertexDetails")
+            .field("lazy", &self.lazy)
+            .field("cached", &self.cached)
+            .finish()
+    }
+}
+
+impl<V> AsAny for HybridVertexDetails<V>
+where
+    V: StoreVertex + 'static,
+{
+    fn as_any_mut(&mut self) -> &mut dyn Any {
+        self
+    }
+
+    fn as_any_ref(&self) -> &dyn Any {
+        self
+    }
+}
+
+impl<V> Drop for HybridVertexDetails<V>
+where
+    V: StoreVertex + 'static,
+{
+    fn drop(&mut self) {
+        let ptr = self.lazy.load(Ordering::SeqCst);
+        if !ptr.is_null() {
+            unsafe {
+                std::ptr::drop_in_place(ptr);
+            }
+        }
+    }
+}
+
+impl<V> Details for HybridVertexDetails<V>
+where
+    V: StoreVertex + 'static,
+{
+    fn get_property(&self, key: &NameOrId) -> Option<PropertyValue> {
+        if let NameOrId::Id(key) = key {
+            if let Some(ref cached) = self.cached {
+                cached
+                    .get(&(*key as PropId))
+                    .map(|obj| PropertyValue::Owned(obj.clone()))
+            } else if let Some(ptr) = self.get_vertex_ptr() {
+                unsafe {
+                    (*ptr)
+                        .get_property(*key as PropId)
+                        .map(|prop| PropertyValue::Owned(encode_runtime_prop_val(prop)))
+                }
+            } else {
+                None
+            }
+        } else {
+            info!("Have not support getting property by prop_name in gs_store yet");
+            None
+        }
+    }
+
+    fn get_all_properties(&self) -> Option<HashMap<NameOrId, Object>> {
+        // note that column filter is pushed down, so just return all props
+        let lazy_props: Option<HashMap<NameOrId, Object>> = if let Some(ptr) = self.get_vertex_ptr() {
```

Comment:
1. If props.is_empty(), just return an empty HashMap instead of None, as identical to what other XXDetails does.
2. Then no need to `match (lazy_props, cached_props)`, `lazy_props.extend(cached_props)` is okay.